src/App.tsx

```diff
@@ -1,33 +1,194 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import viteLogo from '/vite.svg'
+import { useReducer } from 'react'
 import './App.css'
 
+
+export const ACTIONS = {
+  ADD_DIGIT: 'add-digit',
+  CHOOSE_OPERATION: 'choose-operation',
+  CLEAR: 'clear',
+  DELETE_DIGIT: 'delete-digit',
+  EVALUATE: 'evaluate',
+}
+
+const DigitButton = ({ dispatch, digit }) => {
+  return (
+    <button 
+    onClick={() => dispatch({ type: ACTIONS.ADD_DIGIT, payload: { digit } })}
+     > 
+     {digit}
+    </button>
+    )
+}
+
+const OperationButton = ({ dispatch, operation }) => {
+  return (
+    <button 
+    onClick={() => dispatch({ type: ACTIONS.CHOOSE_OPERATION, payload: { operation } })}
+     > 
+     {operation}
+    </button>
+    )
```

Comment:
🔴 Example red comment: serious issue, code not according to criteria, show-stopper, blocks a PR's merge